### PR TITLE
Remove unneeded promise constructors

### PR DESCRIPTION
**Pull request by JPinkney**
Signed-off-by: Josh Pinkney <joshpinkney@gmail.com>

### What does this PR do?
This PR removes promise constructors since they aren't actually used anywhere. We've always just used Promise and never passed anything different in

### Is it tested? How?

npm run test

### CODE REVIEW

**Review by evidolob — APPROVED** [review]

File: src/languageservice/services/yamlHover.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
  *--------------------------------------------------------------------------------------------*/
 'use strict';
 
-import { PromiseConstructor, Thenable } from 'vscode-json-languageservice';
+import { Thenable } from 'vscode-json-languageservice';
```
Comment (joshuawilson) [inline src/languageservice/services/yamlHover.ts:8]: I don't think you need `Thenable` either.

File: src/languageservice/services/yamlCompletion.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { ASTNode, ObjectASTNode, PropertyASTNode } from '../jsonASTTypes';
 import { parse as parseYAML } from '../parser/yamlParser07';
 import { YAMLSchemaService } from './yamlSchemaService';
 import { JSONSchema, JSONSchemaRef } from '../jsonSchema';
-import { PromiseConstructor, Thenable, JSONWorkerContribution, CompletionsCollector } from 'vscode-json-languageservice';
+import { Thenable, CompletionsCollector } from 'vscode-json-languageservice';
```
Comment (joshuawilson) [inline src/languageservice/services/yamlCompletion.ts:13]: Don't need `Thenable` here either.

**Comment by joshuawilson** [comment]
I guess I can stop reviewing this. :-)

**Comment by JPinkney** [comment]
I'm going to make another PR that switches all Thenable to Promise :smile:

**Comment by joshuawilson** [comment]
You promise?

**Pull request merged**